app/code/Magento/Catalog/Block/Adminhtml/Product/Attribute/Set/Main/Formset.php

```diff
@@ -43,7 +50,11 @@ protected function _prepareForm()
 
         /** @var \Magento\Framework\Data\Form $form */
         $form = $this->_formFactory->create();
-        $fieldset = $form->addFieldset('set_name', ['legend' => __('Edit Attribute Set Name')]);
+        if (!$this->getRequest()->getParam('id', false)) {
```

Comment:
Please consider to move this if logic to the separate method:
```
$fieldset = $form->addFieldset('set_name', ['legend' => $this->getAttributeSetLabel()]);
```
```
    private function getAttributeSetLabel()
    {
        if ($this->getRequest()->getParam('id', false)) {
            return __('Edit Attribute Set Name');
        }

        return __('Attribute Set Information');
    }
```

Reply:
I changed the source code. Please check it for me :)  @Stepa4man